File: src/com/galinasoft/simucovid/Individu.java
package com.galinasoft.simucovid;

public class Individu {
	final static int MAXAGESTUDENT = 30; // tune it for specify % of young people
	final static int MINAGESENIOR = 65; // tune it for specify % of young people
	short age;
	int school;
	int enterprise;
	int family;
	int friendGroup;
	boolean alive;
	boolean infected;
	short howManyInfected;
	HowInfected howInfected; // 1=family 2=Work; 3=school
	boolean immunized;
	short dayOfContamination; 
	short virusGeneration;
	
	public Individu(short age, int school, int work, int family, int friendGroup) {
		this.age = age;
		this.school = school;
		this.enterprise = work;
		this.family = family;
		this.friendGroup = friendGroup;
		this.alive = true;
		this.infected = false;
		this.immunized = false;
		this.dayOfContamination = -1;
		this.virusGeneration = 0;
		this.howManyInfected = 0;
	}
	
	public boolean isWorking() { // student or adult not senior
		return ((age % 4 < 3) && !isSenior()); // 75% des actifs travaillent ou vont à l'école mais pas les seniors
	}
	
	public boolean isSenior() {
		return age >= MINAGESENIOR;  // don't tune it for specify % of senior people, use
	}
	
	public boolean isStudent() {
		return age < MAXAGESTUDENT;  
	}
	
	public boolean isContagious(int day) {
		return (infected && !immunized && alive && ((day - dayOfContamination) > Simu.dayStartOfContagiousness));
	}
	
	public boolean isInfected() {
		return infected ;
	}
	
	public boolean isInfectable() {
		return !infected && !immunized && alive;
	}
	public boolean isDetected(int day) {
		return (alive && infected && (day - dayOfContamination) > Simu.dayOfFirstSign);
	}
}

File: python/Individu.py
from Simucovid import Simu

class Individu:
	MAXAGESTUDENT = 30 # tune it for specify % of young people
	MINAGESENIOR = 65 # tune it for specify % of young people
	
	def __init__(self, age, school, work, family, friendGroup):
		self.age = age
		self.school = school
		self.enterprise = work
		self.family = family
		self.friendGroup = friendGroup
		self.alive = True
		self.infected = False
		self.immunized = False
		self.dayOfContamination = -1
		self.howInfected = -1
	
	def isWorking(self):
		return ((self.age % 4 < 3) and not self.isSenior())
	
	
	def isSenior(self):
		return self.age >= self.MINAGESENIOR
	
	
	def isStudent(self):
		return self.age <= self.MAXAGESTUDENT
	
	
	def isContagious(self):
		return (self.alive and self.infected and not self.immunized)
	
	def isInfectable(self):
		return self.alive and not self.infected and not self.immunized
	
	def isDetected(self, day):
		return (self.alive and self.infected and (day - self.dayOfContamination) > Simu.DayOfFirstSign)
	

